byron/chain/executable-spec/test/Cardano/AbstractSize/Properties.hs

```diff
@@ -0,0 +1,163 @@
+      , _bProtVer = ProtVer
+                      { _pvMaj = 0
+                      , _pvMin = 1
+                      , _pvAlt = 1
+                      }
+      }
+
+aBlock :: Block
+aBlock = Block { _bHeader = aHeader
+               , _bBody = aBody
+               }
+
+-- | A BlockHeader term has fixed typeReps
+exampleTypeRepsBlockHeader :: Assertion
+exampleTypeRepsBlockHeader =
+  (typeReps aHeader) @?= Seq.fromList [ typeOf (undefined::BlockHeader),
+        typeOf (undefined::Hash), typeOf (undefined::Hash),
+        typeOf (undefined::Hash), typeOf (undefined::Hash),
+        typeOf (undefined::Slot), typeOf (undefined::Word64),
+        typeOf (undefined::VKey), typeOf (undefined::Owner),
+        typeOf (undefined::Natural), typeOf (undefined::Sig Hash) ]
+
+-- | A BlockBody has variable typeReps, depending on the collections
+-- [DCert], [TxWits], [Vote] and [STag] (in UProp)
+--
+--   In this example, we can see the repetition of typeReps for 2 TxWits
+exampleTypeRepsBlockBody :: Assertion
+exampleTypeRepsBlockBody =
+     (typeReps aBody) @?=
+        Seq.fromList [
+            typeOf (undefined::BlockBody)
+          , typeOf (undefined::[DCert])
+          , typeOf (undefined::[TxWits]) ]
+        >< (typeReps aTxWits)
+        >< (typeReps aTxWits)
+        >< Seq.fromList [
+              typeOf (undefined::Maybe UProp), typeOf (undefined::[Vote])
+            , typeOf (undefined::ProtVer), typeOf (undefined::Natural)
+            , typeOf (undefined::Natural), typeOf (undefined::Natural) ]
+
+-- | The typeReps for a 'Block' is a combination of typeReps for
+-- the header and body in the block.
+exampleTypeRepsBlock :: Assertion
+exampleTypeRepsBlock =
+     (typeReps aBlock) @?= typeOf (undefined::Block)
+                           <| typeReps aHeader
+                           >< typeReps aBody
+
+--------------------------------------------------------------------------------
+-- Properties of abstractSize for Block/Header/Body
+--------------------------------------------------------------------------------
+
+-- | Make a singleton cost of "1" for the given term's type
+mkCost :: forall a. Typeable a => Map TypeRep Size
+mkCost = Map.singleton (typeOf (undefined::a)) 1
+
+-- | This property tests that for abstractSize
+--   - we should account for each DCert/TxWits/Vote/UProp/STag in a Block Body
+--   - the BlockHeader and BlockBody should each be counted only once
+propMultipleOfSizesBlock
+  :: MonadTest m => Block -> m ()
+propMultipleOfSizesBlock b =
+  let
+    body_ = (_bBody b)
+  in do
+    abstractSize (mkCost @DCert)  b === length (_bDCerts body_)
+    abstractSize (mkCost @TxWits) b === length (_bUtxo body_)
+    abstractSize (mkCost @Vote)   b === length (_bUpdVotes body_)
+    abstractSize (mkCost @UProp)  b === length (maybeToList (_bUpdProp body_))
+    abstractSize (mkCost @STag)   b
+      === case _bUpdProp body_ of
+            Just uprop -> Set.size (_upSTags uprop)
+            Nothing -> 0
```

Comment:
Right now `CHAIN` has no update payload, so we only test the `Nothing` case. However it is nice that we have this here for when we add the update payload.